mlir/lib/Transforms/lab4/bozin_dmitry/BozinCeilDiv.cpp

```diff
@@ -0,0 +1,72 @@
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+
+using namespace mlir;
+
+namespace {
+
+enum class CeilDivType { Signed, Unsigned };
+
+class CeilDivPass
+    : public PassWrapper<CeilDivPass, OperationPass<LLVM::LLVMFuncOp>> {
+public:
+  StringRef getArgument() const final { return "bozin_ceildiv"; }
+  StringRef getDescription() const final {
+    return "breaks arith.ceildivui and arith.ceildivsi operations into arith. "
+           "calculations";
+  }
+
+  void runOnOperation() override {
+    getOperation()->walk([&](Operation *op) {
+      if (auto ceilDivUI = dyn_cast<arith::CeilDivUIOp>(op)) {
+        transformCeilDiv(ceilDivUI, CeilDivType::Unsigned);
+      } else if (auto ceilDivSI = dyn_cast<arith::CeilDivSIOp>(op)) {
+        transformCeilDiv(ceilDivSI, CeilDivType::Signed);
+      }
+    });
+  }
+
+private:
+  template <typename CeilDivOp>
+  void transformCeilDiv(CeilDivOp op, CeilDivType divType) {
```

Comment:
Please don't use `divType` argument as there is enough information for that already in first argument.